tools/bash/runfiles/runfiles.bash

```diff
@@ -357,7 +357,25 @@ function runfiles_rlocation_checked() {
     if [[ "${RUNFILES_LIB_DEBUG:-}" == 1 ]]; then
       echo >&2 "INFO[runfiles.bash]: rlocation($1): looking in RUNFILES_MANIFEST_FILE ($RUNFILES_MANIFEST_FILE)"
     fi
-    local -r result=$(__runfiles_maybe_grep -m1 "^$1 " "${RUNFILES_MANIFEST_FILE}" | cut -d ' ' -f 2-)
+    # If the rlocation path contains a space or newline, it needs to be prefixed
+    # with a space and spaces, newlines, and backslashes have to be escaped as
+    # \s, \n, and \b.
+    if [[ "$1" == *" "* || "$1" == *$'\n'* ]]; then
+      local search_prefix=" $(echo -n "$1" | sed 's/\\/\\b/g; s/ /\\s/g')"
```

Comment:
you forgot to also escape newlines here (but you check whether there are newlines in the runfiles path in the line  just above?)

Reply:
Newlines are escaped in the line below. Escaping newlines with `sed` requires commands other than `s`, which I found to be too hard to read. I could replace the `sed` usage with Bash variable substitution entirely, if you prefer that.

Reply:
Nevermind. I overlooked that. You're good to go.